app/javascript/groceries/components/DeletedItemToast.vue

```diff
@@ -0,0 +1,26 @@
+<template lang="pug">
+div
+  | You deleted '{{ deletedItemName }}'.
+  |
+  button.btn-primary(@click='restoreItem') Undo
+</template>
+
+<script setup lang="ts">
+import { http } from '@/shared/http';
+
+const props = defineProps({
+  deletedItemName: {
+    type: String,
+    required: true,
+  },
+
+  restoreItemPath: {
+    type: String,
+    required: true,
+  },
+});
+
+function restoreItem() {
+  http.post(props.restoreItemPath);
```

Comment:
I think I have previously made an effort to get all HTTP requests out of Vue components and into stores. Here, we are making an HTTP request in a Vue component. This feels better to me than the indirection of moving this into the store, just for the sake of following that principle. I think it's probably a good general principle, but I think this is a case where it doesn't make a lot of sense, particularly since there is no direct effect on / interaction with the data that is in the groceries Pinia store.